index.js

```diff
@@ -10,7 +10,10 @@ module.exports = {
           * Always delay requires, otherwise your plugin will cause trouble
           * with db-migrates performance and generates issues to your users.
           */
-        require('typescript').register();
+        var typescriptModule = require('typescript');
+        if (typescriptModule && typescriptModule.register) {
```

Comment:
Well I would use try catch only if really necessary. And in this case I probably would want an error to be thrown or at least being displayed.
A plugin that just silently do not work is always more confusing for a user than a plugin clearly stating that it does not work.

So I just wonder: `typescriptModule.register`, if you say this wouldn't exist with `ES6` anymore. Why would the author of the typescriptModule remove it? That does not make sense for me, as even if there is `ES6` there is no reason to change the API.